core/src/main/scala/com/nvidia/spark/rapids/tool/ToolTextFileWriter.scala

```diff
@@ -34,55 +35,45 @@ class ToolTextFileWriter(
     finalLocationText: String,
     hadoopConf: Option[Configuration] = None) extends Logging {
 
-  // use same as Spark event log writer
-  val LOG_FILE_PERMISSIONS = new FsPermission(Integer.parseInt("660", 8).toShort)
-  val LOG_FOLDER_PERMISSIONS = new FsPermission(Integer.parseInt("770", 8).toShort)
-  private val textOutputPath = new Path(s"$finalOutputDir/$logFileName")
-  private val hadoopConfToUse = hadoopConf.getOrElse(RapidsToolsConfUtil.newHadoopConf)
-
-  private val defaultFs = FileSystem.getDefaultUri(hadoopConfToUse).getScheme
-  private val isDefaultLocal = defaultFs == null || defaultFs == "file"
-  private val uri = textOutputPath.toUri
+  private val textOutputLoc = s"$finalOutputDir/$logFileName"
 
-  def getFileOutputPath: Path = textOutputPath
+  def getFileOutputPath: Path = new Path(textOutputLoc)
 
   // The Hadoop LocalFileSystem (r1.0.4) has known issues with syncing (HADOOP-7844).
   // Therefore, for local files, use FileOutputStream instead.
   // this overwrites existing path
-  private var outFile: Option[FSDataOutputStream] = {
-    val fs = FileSystem.get(uri, hadoopConfToUse)
-    val outStream = if ((isDefaultLocal && uri.getScheme == null) || uri.getScheme == "file") {
-      FileSystem.mkdirs(fs, new Path(finalOutputDir), LOG_FOLDER_PERMISSIONS)
-      Some(new FSDataOutputStream(new FileOutputStream(uri.getPath), null))
-    } else {
-      Some(fs.create(textOutputPath))
+  private var utf8Writer: Option[BufferedWriter] = {
```

Comment:
nit: This comment might not be relevant here any more since we use `BufferedWriter` now.

Reply:
We are still using the FileOutputStream: `new FSDataOutputStream(new FileOutputStream(uri.getPath))`
The new changes wraps the `FSDataOutputStream` inside `BuffereWriter` for encoding.